### PR TITLE
refactor: convert BrowserController to TS

### PR DESCRIPTION
**Pull request by vladfrangu**
WIP: depends on #35, will rebase once thats merged

### CODE REVIEW

**Comment by vladfrangu** [comment]
CC: @pocesar @B4nan @mnmkng

File: src/events.ts [hunk 1]
```diff
@@ -5,8 +5,8 @@ export const BROWSER_POOL_EVENTS = {
 
     PAGE_CREATED: 'pageCreated',
     PAGE_CLOSED: 'pageClosed',
-};
+} as const;
```
Comment (B4nan) [inline src/events.ts:8]: those might be better of done as enums i guess

**Review by B4nan — APPROVED** [review]
the generics are getting quite complex, it looks good, but we need to ensure if it does what it should once we are done with the code migration (i guess once we have the tests migrated to TS it will be enough)

**Comment by vladfrangu** [comment]
> the generics are getting quite complex, it looks good, but we need to ensure if it does what it should once we are done with the code migration (i guess once we have the tests migrated to TS it will be enough)

I know, but sadly they are needed for strict typings.. 😔 

However, this might give you some re-assurance (and I've done mini tests like these to make sure it got everything automatically):

image
image

Also, I noticed the fact the `typeof chromium` was erroring, I have fixed that (was an issue on my part)

**Pull request merged**